### DISCUSSION TITLE
use autokey to control gesture with logitech master MX

### DISCUSSION THREAD

**Discussion by popeliao** [autokey/autokey#687] [Q&A]
Hi folks,

Autokey is pretty cool!
I am wondering has anyone tried using autokey to control a logitech master MX mouse in linux?

On mac/OS, they have logitech option software to have complicated gestures, e.g. I can push the middle key and swipe right to go the next tab in chrome.

**Comment by josephj11** [comment]
TL;DR: It's on our wish list, but not likely any time soon.

Up until now, mouse support has been pretty limited. For quite a while it didn't work at all. More recently, support was added for "teleporting" the mouse to an absolute or relative location and clicking a button.

With 0.96, currently in beta testing, we now support moving and dragging the mouse. This is the base capability required for any gesture based events.

I've been meaning to try it out, but haven't had the time yet. Since this is a new capability, we need as many people as possible to test it.

The beta has been out for a number of months and a small number of us (including me) are using it on a daily basis with almost no issues, so it seems pretty stable. You can find it [here](github.com/autokey/autokey/releases/tag/v0.96.0-beta.10).

If you're not using a Debian derivative OS, then use these [instructions](https://github.com/autokey/autokey/wiki/Installing#pip-installation) for installing with pip.

Be sure to make backups of all your scripts and phrases and their hidden metadata sidecar files before installing the beta. This is particularly important with this release because the sidecar files are no longer hidden and will be automatically renamed by the beta (removing the leading dot from their names), so you would have to manually rename them back to their old names or symlink them if you wanted to revert to 0.95.10 or earlier.

As for your specifics, I use a touchpad, so I'm not particularly familiar with mice. If you want to implement a gesture that some receiving application window recognizes, you will have to manually figure out the associated mouse movements and relative coordinates and program them in a script. 

There is currently no way to trigger an AutoKey event with any mouse behavior, so your script would still need to be activated by a hotkey or trigger abbreviation.

Issues #180 and #519 would be steps toward what you want, but you can file one or more enhancement request issues if you want to. If you do, please give us as much information as possible as to exactly how you would like it to work.

Our developer resources are quite limited at the moment, so significant enhancements (even very nice ones) are relatively low priority.

Feel free to join us on our support forums to discuss such things. See boiler plate below. (Posting things like this here is fine, but more people will see them there.)

----

BTW: Whenever you have a question (as opposed to an issue) it's better to post it on our [list](https://groups.google.com/forum/#!forum/autokey-users) or on [Gitter](https://gitter.im/autokey/autokey). The list is more user oriented whereas more technical questions will get a better answer on Gitter.

Both of these venues have a number of active users, so you are likely to get more, better, and faster answers on them than here on GitHub where only our very busy developers and a few users see the posts. We also prefer to have the developers developing rather than taking time to answer questions and manage issues that the user community could handle for them.
